bundles/specmate-persistency-cdo/src/com/specmate/persistency/cdo/internal/CDOPersistencyService.java

```diff
@@ -281,19 +284,39 @@ public IView openView() {
 		return view;
 	}
 
-	private boolean readConfig(Config config) {
-		jdbcConnection = config.cdoJDBCConnection();
-		repository = config.cdoRepositoryName();
-		resourceName = config.cdoResourceName();
-		userResourceName = config.cdoUserResourceName();
-		if (StringUtils.isEmpty(repository) || StringUtils.isEmpty(resourceName)
-				|| StringUtils.isEmpty(userResourceName)) {
+	private boolean readConfig(Config defaultConfig) {
+		try {
+			Dictionary<String, Object> properties = configurationAdmin.getConfiguration(CDOPersistenceConfig.PID).getProperties();

```

Comment:
this is not necessary, the properties obtained here are the same as the parameter provided in the activate method (or should be, otherwise there is an error somewhere else)